ddtrace/tracer/option.go

```diff
@@ -654,11 +658,13 @@ func WithDogstatsdAddress(addr string) StartOption {
 	}
 }
 
-// WithSamplingRules specifies the sampling rates to apply to spans based on the
+// WithSamplingRules specifies the sampling rates to apply to trace spans based on the
```

Comment:
Sorry but this is still not clear. All spans are "trace spans". What do you mean?

Reply:
I think what you have now is already hinting at a good approach. I suggest to keep `WithSamplingRules` and keep the type as a field in the `SamplingRule` as you do today. And allow a rule to refer either to the trace, or a span. Split them inside the `ruleSampler`. That's how I'd do it, WDYT?

Reply:
You could even add a wrapper that converts the already existing sampling rules. For example:
```go
// WithSpanSamplingRule changes the type of sr to SamplingTypeSpan.
func WithSpanSamplingRule(sr SamplingRule) SamplingRule {
    sr.Type = SamplingRuleTypeSpan
    return sr
}
```
And then you could use all the existing rules for spans too:
```go
tracer.Start(tracer.WithSamplingRule(tracer.WithSpanSamplingRule(tracer.NameRule(...))))
```
This is just one idea, and there are probably better ones. I'm just trying to demonstrate possiblities.

Reply:
if we merge into one, the sampler will look like 
` type RuleSampler struct {
	rules []SamplingRule 
	globalRate float64        // we don't care about this for individual span sampling
	limiter    *rateLimiter  // we don't care about this for individual span sampling
}

type SamplingRule struct {
	Service      *regexp.Regexp
	Name         *regexp.Regexp
	Rate         float64
	MaxPerSecond float64 // we don't care about this for regular trace sampling

	exactService string
	exactName    string
	limiter      *rateLimiter // we don't care about this for regular trace sampling
}

`

Reply:
That's fine. You can put them in separate groups (separated by a blank line, and mention this). For the `RulesSampler` struct, it might make sense to have `traceRules` and `spanRules` separated. Might be easier when calling `sampleSpan` or `sampleTrace`.

As for `MaxPerSecond`, we can maybe make it work for traces too?